src/contexts/useFileEncryption.ts

```diff
@@ -0,0 +1,94 @@
+import { useState, useEffect, useCallback } from 'react';
+import { encrypt } from 'ecies-geth';
+import { utils } from 'ethers';
+
+const useFileEncryption = () => {
+  const [file, setFile] = useState<File | null>(null);
+  const [fileByteArray, setFileByteArrayArray] = useState<ArrayBuffer | string | null>(null);
+  const [fileEncryptedRecipient, setFileEncryptedRecipient] = useState<Buffer | null>(null);
+  const [recipientPublicKey, setRecipientPublicKey] = useState<string | null>(null);
+  const [doubleEncryptedFile, setDoubleEncryptedFile] = useState<Buffer | null>(null);
+  const [randomPublicKey, setRandomPublicKey] = useState<string>('');
+
+  function hexToBytes(hex: string, pad = false) {
+    let byteArray = utils.arrayify(hex);
+    if (pad) {
+      let padByte = new Uint8Array([4]);
+      return Buffer.from(new Uint8Array([...padByte, ...byteArray]));
+    } else {
+      return Buffer.from(byteArray);
+    }
+  }
+
+  const firstEncryption = useCallback(async () => {
+    try {
+      //todo - rewrite verify public key to use less code and not substr
+      let formattedPublicKey;
+      if (recipientPublicKey?.substr(0, 4) !== '0x04') {
+        formattedPublicKey = '0x04' + recipientPublicKey;
+      }
```

Comment:
Should this callback return early if `recipientPublicKey` is null or undefined? In its current state, it looks like this conditional would return `true` and `formattedPublicKey` would be set to something like: `'0x04' + null`

Reply:
Yes that is possible, but currently firstEncryption() is only called in the useEffect and it checks that recipientPublicKey is not null before it's called. I don't think we want firstEncryption to throw if the key is null? and just returning early also seems problematic.

Reply:
Ah ok -- then could the `?` be removed from `recipientPublicKey?` -- having the optional chaining parameter there indicates `recipientPublicKey` could be null/undefined, but sounds like it won't be.